shell/platform/fuchsia/flutter/platform_view.h

```diff
@@ -116,9 +116,12 @@ class PlatformView final : public flutter::PlatformView,
   fml::TimeDelta vsync_offset_;
   zx_handle_t vsync_event_handle_ = 0;
 
-  float view_width_ = 0.0f;        // Width in logical pixels.
-  float view_height_ = 0.0f;       // Height in logical pixels.
-  float view_pixel_ratio_ = 0.0f;  // Logical / physical pixel ratio.
+  // Logical / physical pixel ratio.  Defaults to 0 in order to avoid creating
+  // ViewportMetrics until Scenic sends at least one Metrics event.
+  float view_pixel_ratio_ = 0.f;
+
+  float view_width_ = 0.f;   // Width in logical pixels.
+  float view_height_ = 0.f;  // Height in logical pixels.
```

Comment:
Should we maybe just make these `std::optional`s, and then you can get rid of the bool trackers in the impl and just check that they all `has_value` before sending metrics?

Reply:
Done, although I kept the (single) bool tracker for the sake of rejecting duplicate updates